infrastructure/smart-contracts/contracts/upgrades/referee/Referee9A.sol

```diff
@@ -0,0 +1,1333 @@
+    mapping(uint256 => address) public assignedKeyToPool;
+
+    // Mapping for pool to assigned key count for calculating max stake amount
+    mapping(address => uint256) public assignedKeysToPoolCount;
+
+    // The maximum number of NodeLicenses that can be staked to a pool
+    uint256 public maxKeysPerPool;
+
+    // The pool factory contract that is allowed to update the stake state of the Referee
+    address public poolFactoryAddress;
+    
+    // Mapping for amount of assigned keys of a user
+    mapping(address => uint256) public assignedKeysOfUserCount;
+
+    // Challenge number the KYC check was removed for claims
+    uint256 public kycCheckRemovedChallengeNumber;
+    
+    // Mapping to store all of the pool submissions
+    mapping(uint256 => mapping(address => PoolSubmission)) public poolSubmissions;
+
+    /**
+     * @dev This empty reserved space is put in place to allow future versions to add new
+     * variables without shifting down storage in the inheritance chain.
+     * See https://docs.openzeppelin.com/contracts/4.x/upgradeable#storage_gaps
+     */
+    uint256[488] private __gap;
+
+    // Struct for the submissions
+    struct Submission {
+        bool submitted;
+        bool claimed;
+        bool eligibleForPayout;
+        uint256 nodeLicenseId;
+        bytes assertionStateRootOrConfirmData;
+    }
+
+    // Struct for the challenges
+    struct Challenge {
+        bool openForSubmissions; // when the next challenge is submitted for the following assertion, this will be closed.
+        bool expiredForRewarding; // when this is true, this challenge is no longer eligible for claiming
+        uint64 assertionId;
+        bytes32 assertionStateRootOrConfirmData; // Depending on the BOLD 2 deployment, this will either be the assertionStateRoot or ConfirmData
+        uint64 assertionTimestamp; // equal to the block number the assertion was made on in the rollup protocol
+        bytes challengerSignedHash;
+        bytes activeChallengerPublicKey; // The challengerPublicKey that was active at the time of challenge submission
+        address rollupUsed; // The rollup address used for this challenge
+        uint256 createdTimestamp; // used to determine if a node license is eligible to submit
+        uint256 totalSupplyOfNodesAtChallengeStart; // keep track of what the total supply opf nodes is when the challenge starts
+        uint256 rewardAmountForClaimers; // this is how much esXai should be allocated to the claimers
+        uint256 amountForGasSubsidy; // this is how much Xai was minted for the gas subsidy
+        uint256 numberOfEligibleClaimers; // how many submitters are eligible for claiming, used to determine the reward amount
+        uint256 amountClaimedByClaimers; // keep track of how much Xai has been claimed by the claimers, primarily used to expire unclaimed rewards 
+    }
+
+    // Struct for PoolSubmissions
+    struct PoolSubmission {
+        bool submitted;
```

Comment:
Missing pool address?

Reply:
There's no need to store the pool address as the pool address is part of the mapping:

`    // Mapping to store all of the pool submissions
    mapping(uint256 => mapping(address => PoolSubmission)) public poolSubmissions;`